### ISSUE TITLE
TFLite representative_dataset

### ISSUE THREAD

**Issue by lissyx** [mozilla/DeepSpeech#2416]
We can feed TFLite converter a `representative_dataset` that helps for optimization / quantization process.

**Comment by lissyx** [comment]
We could use the training test set (complete ? reduced ?) for that purpose ? @kdavis-mozilla @reuben @tilmankamp

**Comment by kdavis-mozilla** [comment]
Shouldn't we use the dev set if we're being honest?

**Comment by lissyx** [comment]
I would expect more bias from the dev set than the test set for that purpose, wouldn't we?

**Comment by kdavis-mozilla** [comment]
I guess I'm viewing this under the category of hyperparameters which should be optimized with the dev set.

**Comment by lissyx** [comment]
> I guess I'm viewing this under the category of hyperparameters which should be optimized with the dev set.

In this context, that would mean we would have to perform a TFLite export with this `RepresentativeDataset`, compute test set on this TFLite model, right ?

My view here is that we do

**Comment by lissyx** [comment]
> I guess I'm viewing this under the category of hyperparameters which should be optimized with the dev set.

Right, but I'm not sure we should act this way.

**Comment by kdavis-mozilla** [comment]
Why?

**Comment by lissyx** [comment]
Well, current we optimize the WER of the model that we export and then convert to TFLite. So, in this process, IMHO it makes more sense that the representative data set being used for the conversion is the test set, which is already supposed to be representative data set to evaluate model performance.

**Comment by kdavis-mozilla** [comment]
We optimize hyperparameters on the _dev_ set not the _test_ set.

We shouldn't optimize anything on the test set as any results we'd get then would be dishonest.

**Comment by lissyx** [comment]
> We optimize hyperparameters on the _dev_ set not the _test_ set.
> 
> We shouldn't optimize anything on the test set as any results we'd get then would be dishonest.

Oh, I get your point. I think you are right, since TOCO will use this to **optimize** we should definitively use the dev set.

**Comment by lissyx** [comment]
FTR, the `r1.14` code is not supporting custom ops, but `r1.15` seems to. Still a lot of work to get something decent, but for now it looks better than it used to.

**Comment by lissyx** [comment]
There's mostly nothing to make or from this feature, for now.
